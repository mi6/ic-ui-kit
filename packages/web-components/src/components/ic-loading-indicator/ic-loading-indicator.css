@import "../../global/normalize.css";

:host {
  /**
   * @prop --circular-diameter: Diameter of the circular-indicator
   */

  display: block;

  --linear-border-radius: 0.25rem;
  --inner-color: var(--ic-action-default);
  --outer-color: var(--ic-architectural-100);
  --label-color: var(--ic-color-primary-text);
  --margin: none;
  --linear-line-height: var(--ic-space-xs);
}

:host(.light) {
  --inner-color: var(--ic-architectural-white);
  --outer-color: var(--ic-architectural-800);
  --label-color: var(--ic-architectural-white);
}

.ic-loading-container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

:host([size="small"]) {
  --circular-diameter: 2.5rem;
  --linear-line-height: var(--ic-space-xxs);
}

:host([size="default"]) {
  --circular-diameter: 5rem;
}

:host([size="large"]) {
  --circular-diameter: 7.5rem;
}

:host([size="icon"]) {
  display: inline-block;

  --margin: var(--ic-space-xxxs);
  --circular-diameter: 1.25rem;
}

:host(.label) {
  --margin: 0 0 var(--ic-space-xs) 0;
}

.ic-loading-label {
  white-space: nowrap;
  color: var(--label-color);
  margin-left: var(--label-margin-left);
}

.ic-loading-circular-outer {
  height: var(--circular-diameter);
  width: var(--circular-diameter);
}

.ic-loading-circular-outer.indeterminate {
  animation: circular-animation 1s linear;
  animation-iteration-count: infinite;
}

@keyframes circular-animation {
  100% {
    transform: rotate(360deg);
  }
}

.ic-loading-circular-outer,
.ic-loading-linear-outer {
  margin: var(--margin);
}

.ic-loading-linear-outer {
  background-color: var(--outer-color);
  height: var(--linear-line-height);
  width: 100%;
  border-radius: var(--linear-border-radius);
  overflow: hidden;
}

.ic-loading-linear-inner {
  position: relative;
  height: 100%;
  background-color: var(--inner-color);
  border-radius: var(--linear-border-radius);
}

.indeterminate > .ic-loading-linear-inner {
  animation: linear-animation 2s infinite;
}

.determinate > .ic-loading-linear-inner {
  transition: width 0.5s;
  width: var(--linear-width);
}

:host(.full-width) {
  --linear-border-radius: none;
}

@keyframes linear-animation {
  0% {
    width: 0%;
    left: -1%;
  }

  25% {
    width: 0%;
    left: -1%;
  }

  50% {
    width: 101%;
    left: -1%;
  }

  75% {
    width: 0%;
    left: 101%;
  }

  100% {
    width: 0%;
    left: 101%;
  }
}

.ic-loading-circular-svg {
  position: relative;
  width: 100%;
  height: 100%;
  transform: rotate(-90deg);
}

.ic-loading-circular-svg circle {
  width: 100%;
  height: 100%;
  fill: none;
  stroke: var(--outer-color);
  stroke-width: var(--circular-line-width);
  stroke-linecap: round;
}

/* --stroke-dash-array calculated via dashArray method */
.ic-loading-circular-svg circle:nth-child(2) {
  --circular-indeterminate: calc(
    (0.25 * var(--stroke-dasharray)) - var(--stroke-dasharray)
  );

  stroke-dasharray: var(--stroke-dasharray), var(--stroke-dasharray);
  stroke-dashoffset: var(--stroke-dashoffset, var(--circular-indeterminate));
  stroke: var(--compact-step-inner-color, var(--inner-color));
}

:host(.inner-label) .ic-loading-circular-inner {
  display: grid;
}

:host(.inner-label) .ic-loading-circular-svg,
.inner-text {
  grid-column: 1;
  grid-row: 1;
}

:host(.not-required.compact-step-progress-indicator)
  .ic-loading-circular-svg
  circle:nth-child(2) {
  stroke: var(--ic-architectural-300);
}

.inner-text {
  display: flex;
  align-items: center;
  justify-content: center;
  width: var(--ic-space-lg);
  height: inherit;
  overflow: hidden;
  margin: auto;
  color: var(--ic-status-info);
}

:host(.not-required.compact-step-progress-indicator) .inner-text {
  color: var(--ic-color-tertiary-text);
}

/** High Contrast **/
@media (forced-colors: active) {
  .indeterminate > .ic-loading-circular-inner {
    forced-color-adjust: none;
  }

  .ic-loading-linear-outer {
    border: var(--ic-hc-border);
  }

  .ic-loading-linear-inner {
    background-color: canvastext;
  }

  .ic-loading-circular-svg circle {
    stroke: Background;
  }

  .ic-loading-circular-svg circle:nth-child(2) {
    stroke: canvastext;
  }

  :host(.not-required.compact-step-progress-indicator)
    .ic-loading-circular-svg
    circle:nth-child(2) {
    stroke: GrayText;
  }
}
