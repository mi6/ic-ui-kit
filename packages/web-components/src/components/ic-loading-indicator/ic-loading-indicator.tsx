import { Component, Element, h, Host, Prop, State, Watch } from "@stencil/core";
import {
  IcLoadingCircleXYR,
  IcLoadingSizes,
  IcLoadingTypes,
} from "./ic-loading-indicator.types";
import {
  IcThemeForegroundEnum,
  IcThemeForegroundNoDefault,
} from "../../utils/types";

@Component({
  tag: "ic-loading-indicator",
  styleUrl: "ic-loading-indicator.css",
  shadow: true,
})
export class LoadingIndicator {
  private circularMeter: SVGCircleElement;
  private innerElement?: HTMLDivElement;
  private interval: ReturnType<typeof setInterval>;
  private labelList: string[];
  private outerElement?: HTMLDivElement;

  @Element() host: HTMLIcLoadingIndicatorElement;

  @State() circularDiameter: number;
  @State() circularLineWidth: number;
  @State() indeterminate: boolean;
  @State() indicatorLabel: string;
  @State() showSecond: boolean = false;

  /**
   * The appearance of the loading indicator, e.g. dark or light.
   */
  @Prop() appearance?: IcThemeForegroundNoDefault = "dark";

  /**
   * The description that will be set as the aria-label of the loading indicator when not using a visible label.
   */
  @Prop() description?: string = "Loading";

  /**
   * If `true`, when linear, the full-width variant (i.e. without a border radius) will be displayed.
   */
  @Prop({ reflect: true }) fullWidth?: boolean = false;

  /**
   * @internal The step number of a compact step, managed by ic-step.
   */
  @Prop() innerLabel?: number;

  /**
   * The time in milliseconds before the label changes.
   */
  @Prop() labelDuration?: number = 8000;

  /**
   * The maximum value that the progress value can take.
   * Used to calculate the proportional width of the progress bar.
   */
  @Prop() max?: number = 100;

  /**
   * The minimum value that the progress value can take.
   * Used to calculate the proportional width of the progress bar.
   */
  @Prop() min?: number = 0;

  /**
   * The size of the loading indicator.
   */
  @Prop({ reflect: true }) size?: IcLoadingSizes = "default";

  /**
   * The type of indicator, either linear or circular.
   */
  @Prop({ reflect: true }) type?: IcLoadingTypes = "circular";

  /**
   * The label to be displayed beneath the loading indicator.
   * Display a changing label by separating multiple messages with forward slashes.
   */
  @Prop() label?: string;

  @Watch("label")
  watchPropHandler(): void {
    this.updateLabel();
  }

  /**
   * The current amount of progress made.
   * If not provided, component acts as an indeterminate loading indicator.
   */
  @Prop() progress?: number;

  @Watch("progress")
  watchProgressHandler(): void {
    if (this.type === "circular") {
      this.updateCircularProgressMeter();
    }
  }

  disconnectedCallback(): void {
    clearInterval(this.interval);
  }

  componentWillLoad(): void {
    this.indeterminate = this.progress === undefined;
    this.updateLabel();
  }

  componentDidLoad(): void {
    if (this.type === "circular") {
      this.setCircleLineWidth();
      this.circularMeter = this.host.shadowRoot.querySelector(
        ".ic-loading-circular-svg circle:nth-child(2)"
      );
      this.updateCircularProgressMeter();
    }

    if (Number(this.progress) >= 0 && this.type === "linear") {
      this.setLinearDeterminateWidth();
    }
  }

  componentWillUpdate(): void {
    this.indeterminate = this.progress === undefined;
  }

  componentDidUpdate(): void {
    if (Number(this.progress) >= 0 && this.type === "linear") {
      this.setLinearDeterminateWidth();
    }
  }

  private updateCircularProgressMeter = () => {
    if (!this.indeterminate) {
      this.circularMeter.style.setProperty(
        "--progress-value",
        String(this.progress)
      );
    }
  };

  private getLabel = (
    labelIndex: number,
    setLabel: (label: string) => void
  ) => {
    return new Promise(() => {
      this.interval = setInterval(() => {
        if (labelIndex < this.labelList.length - 1) {
          labelIndex++;
        } else {
          labelIndex = 0;
        }
        setLabel(this.labelList[labelIndex]);
      }, this.labelDuration);
    });
  };

  private getLabelVariant = () => {
    let variant: "label" | "h4" | "h2" = "h4";
    const width = this.outerElement?.offsetWidth;
    if (this.size === "small" || (this.type === "circular" && width < 60)) {
      variant = "label";
    } else if (
      this.size === "large" ||
      (this.type === "circular" && width >= 120)
    ) {
      variant = "h2";
    }
    return variant;
  };

  // Sets the circular indicator line width - accounting for the circle size being altered using the CSS custom property
  private setCircleLineWidth = () => {
    const { offsetWidth: width } = this.outerElement;

    const compactStepCircularLineWidth = this.host.classList.contains(
      "compact-step-progress-indicator"
    )
      ? 40
      : 0;
    const toastDismissTimerCircularLineWidth = this.host.classList.contains(
      "toast-dismiss-timer"
    )
      ? 20
      : 0;

    if (
      width ||
      compactStepCircularLineWidth ||
      toastDismissTimerCircularLineWidth
    ) {
      this.circularLineWidth =
        (compactStepCircularLineWidth ||
          toastDismissTimerCircularLineWidth ||
          width) * 0.1;
      this.circularDiameter =
        compactStepCircularLineWidth ||
        toastDismissTimerCircularLineWidth ||
        width;
      this.outerElement.style.setProperty(
        "--circular-line-width",
        `${this.circularLineWidth}px`
      );
    }
  };

  private setLinearDeterminateWidth = () => {
    if (!this.innerElement) return;
    // Ensure progress cannot be out of bounds

    const progress = Math.min(this.max, Math.max(this.min, this.progress));
    const proportion = (progress - this.min) / (this.max - this.min);
    this.showSecond = proportion > 0.5;
    if (this.showSecond) {
      this.innerElement.classList.remove("clip");
    } else {
      this.innerElement.classList.add("clip");
    }
    this.innerElement.style.setProperty(
      "--linear-width",
      `${proportion * 100}%`
    );
  };

  private calcOuterClass = (): string => {
    let cls = `ic-loading-${this.type}-outer`;
    cls += this.indeterminate ? " indeterminate" : " determinate";
    return cls;
  };

  private updateLabel = (): void => {
    if (this.label !== undefined) {
      this.labelList = this.label.split("/");
      const labelIndex = 0;
      this.indicatorLabel = this.labelList[labelIndex];
      if (this.labelList.length > 1) {
        this.getLabel(labelIndex, (label) => {
          this.indicatorLabel = label;
        });
      }
    }
  };

  private setCircleXY = (): IcLoadingCircleXYR => {
    if (this.circularDiameter > 0) {
      const r = this.circularDiameter / 2;
      const x = r;
      const y = r;
      const nextRadius = r - this.circularLineWidth / 2;
      this.setDashSteps(nextRadius);

      return { x, y, r: nextRadius };
    }
    return { x: 0, y: 0, r: 0 };
  };

  private setDashSteps = (radius: number) => {
    const dashArray = 2 * Math.PI * radius;
    const progress = Math.min(Math.max(this.progress, this.min), this.max);
    const proportion = -1 - (progress - this.min) / (this.max - this.min);

    this.circularMeter.style.setProperty(
      "--stroke-dasharray",
      `${dashArray}px`
    );

    if (!this.indeterminate) {
      this.circularMeter.style.setProperty(
        "--circular-steps-max",
        String(this.max)
      );
      this.circularMeter.style.setProperty(
        "--stroke-dashoffset",
        `${proportion * dashArray}px`
      );
    }
  };

  render() {
    const { appearance, label, description, size, fullWidth, innerLabel } =
      this;
    const { x, y, r } = this.setCircleXY();

    return (
      <Host
        class={{
          ["light"]: appearance === IcThemeForegroundEnum.Light,
          ["label"]: !!label,
          ["full-width"]: fullWidth,
          ["inner-label"]: !!innerLabel,
        }}
      >
        <div class="ic-loading-container">
          <div
            ref={(el) => (this.outerElement = el as HTMLDivElement)}
            class={this.calcOuterClass()}
            role="progressbar"
            aria-labelledby={
              this.label && this.size !== "icon" && "ic-loading-label"
            }
            aria-label={description}
            aria-valuenow={this.progress}
            aria-valuemin={this.min}
            aria-valuemax={this.max}
          >
            <div
              ref={(el) => (this.innerElement = el as HTMLDivElement)}
              class={`ic-loading-${this.type}-inner`}
            >
              {this.innerLabel &&
                this.innerLabel !== undefined &&
                this.size === "small" && (
                  <ic-typography variant="subtitle-small" class="inner-text">
                    {this.innerLabel}
                  </ic-typography>
                )}
              {this.type === "circular" && (
                <svg
                  class="ic-loading-circular-svg"
                  viewBox={`0 0 ${this.circularDiameter || 0} ${
                    this.circularDiameter || 0
                  }`}
                >
                  <circle cx={`${x}`} cy={`${y}`} r={`${r}`}></circle>
                  <circle cx={`${x}`} cy={`${y}`} r={`${r}`}></circle>
                </svg>
              )}
            </div>
          </div>
          {label && size !== "icon" && (
            <ic-typography
              id="ic-loading-label"
              class="ic-loading-label"
              role="status"
              variant={this.getLabelVariant()}
            >
              <p>{this.indicatorLabel}</p>
            </ic-typography>
          )}
        </div>
      </Host>
    );
  }
}
