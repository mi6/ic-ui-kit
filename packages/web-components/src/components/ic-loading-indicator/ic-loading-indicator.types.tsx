export type IcLoadingSizes = "default" | "small" | "large" | "icon";

export type IcLoadingTypes = "circular" | "linear";

export interface IcLoadingCircleXYR {
  x: number;
  y: number;
  r: number;
}
