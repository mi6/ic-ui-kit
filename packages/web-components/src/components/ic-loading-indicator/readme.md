# ic-loading



<!-- Auto Generated Below -->


## Properties

| Property        | Attribute        | Description                                                                                                                             | Type                                        | Default      |
| --------------- | ---------------- | --------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------- | ------------ |
| `appearance`    | `appearance`     | The appearance of the loading indicator, e.g. dark or light.                                                                            | `"dark" \| "light"`                         | `"dark"`     |
| `description`   | `description`    | The description that will be set as the aria-label of the loading indicator when not using a visible label.                             | `string`                                    | `"Loading"`  |
| `fullWidth`     | `full-width`     | If `true`, when linear, the full-width variant (i.e. without a border radius) will be displayed.                                        | `boolean`                                   | `false`      |
| `label`         | `label`          | The label to be displayed beneath the loading indicator. Display a changing label by separating multiple messages with forward slashes. | `string`                                    | `undefined`  |
| `labelDuration` | `label-duration` | The time in milliseconds before the label changes.                                                                                      | `number`                                    | `8000`       |
| `max`           | `max`            | The maximum value that the progress value can take. Used to calculate the proportional width of the progress bar.                       | `number`                                    | `100`        |
| `min`           | `min`            | The minimum value that the progress value can take. Used to calculate the proportional width of the progress bar.                       | `number`                                    | `0`          |
| `progress`      | `progress`       | The current amount of progress made. If not provided, component acts as an indeterminate loading indicator.                             | `number`                                    | `undefined`  |
| `size`          | `size`           | The size of the loading indicator.                                                                                                      | `"default" \| "icon" \| "large" \| "small"` | `"default"`  |
| `type`          | `type`           | The type of indicator, either linear or circular.                                                                                       | `"circular" \| "linear"`                    | `"circular"` |


## CSS Custom Properties

| Name                  | Description                        |
| --------------------- | ---------------------------------- |
| `--circular-diameter` | Diameter of the circular-indicator |


## Dependencies

### Used by

 - [ic-button](../ic-button)
 - [ic-menu](../ic-menu)
 - [ic-step](../ic-step)
 - [ic-toast](../ic-toast)

### Depends on

- [ic-typography](../ic-typography)

### Graph
```mermaid
graph TD;
  ic-loading-indicator --> ic-typography
  ic-button --> ic-loading-indicator
  ic-menu --> ic-loading-indicator
  ic-step --> ic-loading-indicator
  ic-toast --> ic-loading-indicator
  style ic-loading-indicator fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------


