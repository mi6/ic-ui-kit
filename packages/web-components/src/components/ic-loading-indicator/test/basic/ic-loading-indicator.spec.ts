import { newSpecPage, SpecPage } from "@stencil/core/testing";
import { LoadingIndicator } from "../../ic-loading-indicator";
import { Typography } from "../../../ic-typography/ic-typography";
import { waitForTimeout } from "../../../../testspec.setup";

let page: SpecPage;

beforeAll(() => {
  jest.spyOn(console, "warn").mockImplementation(jest.fn());
});

afterEach(() => {
  page.setContent("");
});

describe("ic-loading-indicator component", () => {
  it("should render an indeterminate loading indicator with the correct label and set aria-labelledby", async () => {
    page = await newSpecPage({
      components: [LoadingIndicator, Typography],
      html: `<ic-loading-indicator label="IC Loading Indicator Test"></ic-loading-indicator>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render a determinate loading indicator with the correct label and aria", async () => {
    page = await newSpecPage({
      components: [LoadingIndicator, Typography],
      html: `<ic-loading-indicator progress="30" label="IC Loading Indicator Test"></ic-loading-indicator>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should not display a label is the size is 'icon'", async () => {
    page = await newSpecPage({
      components: [LoadingIndicator, Typography],
      html: `<ic-loading-indicator label="IC Loading Indicator Test" size="icon"></ic-loading-indicator>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should set the typography variant for the label to 'label' for the 'small' size", async () => {
    page = await newSpecPage({
      components: [LoadingIndicator, Typography],
      html: `<ic-loading-indicator label="IC Loading Indicator Test" size="small"></ic-loading-indicator>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should set the typography variant for the label to 'h4' for the 'default' size", async () => {
    page = await newSpecPage({
      components: [LoadingIndicator, Typography],
      html: `<ic-loading-indicator label="IC Loading Indicator Test"></ic-loading-indicator>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should set the typography variant for the label to 'h2' for the 'large' size", async () => {
    page = await newSpecPage({
      components: [LoadingIndicator, Typography],
      html: `<ic-loading-indicator label="IC Loading Indicator Test" size="large"></ic-loading-indicator>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should apply the description prop as an aria-label", async () => {
    page = await newSpecPage({
      components: [LoadingIndicator, Typography],
      html: `<ic-loading-indicator description="IC Loading Indicator Test"></ic-loading-indicator>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should be styled correctly by setting the class names for the chosen indicator type", async () => {
    page = await newSpecPage({
      components: [LoadingIndicator, Typography],
      html: `<ic-loading-indicator type="linear"></ic-loading-indicator>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should set the correct aria if min and max are provided", async () => {
    page = await newSpecPage({
      components: [LoadingIndicator, Typography],
      html: `<ic-loading-indicator progress="30" min="10" max="50"></ic-loading-indicator>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should update label after label-duration passed", async () => {
    page = await newSpecPage({
      components: [LoadingIndicator, Typography],
      html: `<ic-loading-indicator label="waiting/still waiting" label-duration="2000"></ic-loading-indicator>`,
    });

    expect(page.rootInstance.indicatorLabel).toEqual("waiting");

    await waitForTimeout(2001);
    await page.waitForChanges();

    expect(page.rootInstance.indicatorLabel).toEqual("still waiting");

    await waitForTimeout(2001);
    await page.waitForChanges();
    expect(page.rootInstance.indicatorLabel).toEqual("waiting");
  });

  it("should render linear determinate progress bar", async () => {
    page = await newSpecPage({
      components: [LoadingIndicator, Typography],
      html: `<ic-loading-indicator progress="30" min="10" max="50" type='linear'></ic-loading-indicator>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render linear indeterminate progress bar", async () => {
    page = await newSpecPage({
      components: [LoadingIndicator, Typography],
      html: `<ic-loading-indicator type='linear'></ic-loading-indicator>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should test updating label and progress props", async () => {
    page = await newSpecPage({
      components: [LoadingIndicator, Typography],
      html: `<ic-loading-indicator label="IC Loading Indicator Test" size="large"></ic-loading-indicator>`,
    });

    expect(page.rootInstance.label).toBe("IC Loading Indicator Test");

    page.root.label = "New label";
    page.root.progress = 20;
    await page.waitForChanges();

    expect(page.rootInstance.label).toBe("New label");
    expect(page.rootInstance.progress).toBe(20);
  });

  it("should test updating progress - linear", async () => {
    page = await newSpecPage({
      components: [LoadingIndicator, Typography],
      html: `<ic-loading-indicator progress="10" min="10" max="50" type='linear'></ic-loading-indicator>`,
    });
    expect(page.rootInstance.progress).toBe(10);
    page.root.progress = 20;
    await page.waitForChanges();

    expect(page.rootInstance.progress).toBe(20);
  });

  it("should test setting circularDiameter", async () => {
    page = await newSpecPage({
      components: [LoadingIndicator, Typography],
      html: `<ic-loading-indicator label="IC Loading Indicator Test"></ic-loading-indicator>`,
    });

    page.root.progress = "20";
    await page.waitForChanges();
    page.rootInstance.circularDiameter = "100";
    await page.waitForChanges();
    page.rootInstance.setCircleXY();
    await page.waitForChanges();

    const circle = page.root.shadowRoot.querySelector("svg circle");
    const x = circle.getAttribute("cx");
    expect(x).toBe("50");
  });
});
