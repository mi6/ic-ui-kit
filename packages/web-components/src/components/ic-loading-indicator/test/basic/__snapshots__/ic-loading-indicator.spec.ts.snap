// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ic-loading-indicator component should apply the description prop as an aria-label 1`] = `
<ic-loading-indicator description="IC Loading Indicator Test" size="default" type="circular">
  <mock:shadow-root>
    <div class="ic-loading-container">
      <div aria-label="IC Loading Indicator Test" aria-valuemax="100" aria-valuemin="0" class="ic-loading-circular-outer indeterminate" role="progressbar">
        <div class="ic-loading-circular-inner">
          <svg class="ic-loading-circular-svg" viewBox="0 0 0 0">
            <circle cx="0" cy="0" r="0"></circle>
            <circle cx="0" cy="0" r="0"></circle>
          </svg>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</ic-loading-indicator>
`;

exports[`ic-loading-indicator component should be styled correctly by setting the class names for the chosen indicator type 1`] = `
<ic-loading-indicator size="default" type="linear">
  <mock:shadow-root>
    <div class="ic-loading-container">
      <div aria-label="Loading" aria-valuemax="100" aria-valuemin="0" class="ic-loading-linear-outer indeterminate" role="progressbar">
        <div class="ic-loading-linear-inner"></div>
      </div>
    </div>
  </mock:shadow-root>
</ic-loading-indicator>
`;

exports[`ic-loading-indicator component should not display a label is the size is 'icon' 1`] = `
<ic-loading-indicator class="label" label="IC Loading Indicator Test" size="icon" type="circular">
  <mock:shadow-root>
    <div class="ic-loading-container">
      <div aria-label="Loading" aria-valuemax="100" aria-valuemin="0" class="ic-loading-circular-outer indeterminate" role="progressbar">
        <div class="ic-loading-circular-inner">
          <svg class="ic-loading-circular-svg" viewBox="0 0 0 0">
            <circle cx="0" cy="0" r="0"></circle>
            <circle cx="0" cy="0" r="0"></circle>
          </svg>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</ic-loading-indicator>
`;

exports[`ic-loading-indicator component should render a determinate loading indicator with the correct label and aria 1`] = `
<ic-loading-indicator class="label" label="IC Loading Indicator Test" progress="30" size="default" type="circular">
  <mock:shadow-root>
    <div class="ic-loading-container">
      <div aria-label="Loading" aria-labelledby="ic-loading-label" aria-valuemax="100" aria-valuemin="0" aria-valuenow="30" class="determinate ic-loading-circular-outer" role="progressbar">
        <div class="ic-loading-circular-inner">
          <svg class="ic-loading-circular-svg" viewBox="0 0 0 0">
            <circle cx="0" cy="0" r="0"></circle>
            <circle cx="0" cy="0" r="0" style="--progress-value: 30;"></circle>
          </svg>
        </div>
      </div>
      <ic-typography class="ic-loading-label ic-typography-h4" id="ic-loading-label" role="status">
        <mock:shadow-root>
          <slot></slot>
        </mock:shadow-root>
        <p>
          IC Loading Indicator Test
        </p>
      </ic-typography>
    </div>
  </mock:shadow-root>
</ic-loading-indicator>
`;

exports[`ic-loading-indicator component should render an indeterminate loading indicator with the correct label and set aria-labelledby 1`] = `
<ic-loading-indicator class="label" label="IC Loading Indicator Test" size="default" type="circular">
  <mock:shadow-root>
    <div class="ic-loading-container">
      <div aria-label="Loading" aria-labelledby="ic-loading-label" aria-valuemax="100" aria-valuemin="0" class="ic-loading-circular-outer indeterminate" role="progressbar">
        <div class="ic-loading-circular-inner">
          <svg class="ic-loading-circular-svg" viewBox="0 0 0 0">
            <circle cx="0" cy="0" r="0"></circle>
            <circle cx="0" cy="0" r="0"></circle>
          </svg>
        </div>
      </div>
      <ic-typography class="ic-loading-label ic-typography-h4" id="ic-loading-label" role="status">
        <mock:shadow-root>
          <slot></slot>
        </mock:shadow-root>
        <p>
          IC Loading Indicator Test
        </p>
      </ic-typography>
    </div>
  </mock:shadow-root>
</ic-loading-indicator>
`;

exports[`ic-loading-indicator component should render linear determinate progress bar 1`] = `
<ic-loading-indicator max="50" min="10" progress="30" size="default" type="linear">
  <mock:shadow-root>
    <div class="ic-loading-container">
      <div aria-label="Loading" aria-valuemax="50" aria-valuemin="10" aria-valuenow="30" class="determinate ic-loading-linear-outer" role="progressbar">
        <div class="clip ic-loading-linear-inner" style="--linear-width: 50%;"></div>
      </div>
    </div>
  </mock:shadow-root>
</ic-loading-indicator>
`;

exports[`ic-loading-indicator component should render linear indeterminate progress bar 1`] = `
<ic-loading-indicator size="default" type="linear">
  <mock:shadow-root>
    <div class="ic-loading-container">
      <div aria-label="Loading" aria-valuemax="100" aria-valuemin="0" class="ic-loading-linear-outer indeterminate" role="progressbar">
        <div class="ic-loading-linear-inner"></div>
      </div>
    </div>
  </mock:shadow-root>
</ic-loading-indicator>
`;

exports[`ic-loading-indicator component should set the correct aria if min and max are provided 1`] = `
<ic-loading-indicator max="50" min="10" progress="30" size="default" type="circular">
  <mock:shadow-root>
    <div class="ic-loading-container">
      <div aria-label="Loading" aria-valuemax="50" aria-valuemin="10" aria-valuenow="30" class="determinate ic-loading-circular-outer" role="progressbar">
        <div class="ic-loading-circular-inner">
          <svg class="ic-loading-circular-svg" viewBox="0 0 0 0">
            <circle cx="0" cy="0" r="0"></circle>
            <circle cx="0" cy="0" r="0" style="--progress-value: 30;"></circle>
          </svg>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</ic-loading-indicator>
`;

exports[`ic-loading-indicator component should set the typography variant for the label to 'h2' for the 'large' size 1`] = `
<ic-loading-indicator class="label" label="IC Loading Indicator Test" size="large" type="circular">
  <mock:shadow-root>
    <div class="ic-loading-container">
      <div aria-label="Loading" aria-labelledby="ic-loading-label" aria-valuemax="100" aria-valuemin="0" class="ic-loading-circular-outer indeterminate" role="progressbar">
        <div class="ic-loading-circular-inner">
          <svg class="ic-loading-circular-svg" viewBox="0 0 0 0">
            <circle cx="0" cy="0" r="0"></circle>
            <circle cx="0" cy="0" r="0"></circle>
          </svg>
        </div>
      </div>
      <ic-typography class="ic-loading-label ic-typography-h2" id="ic-loading-label" role="status">
        <mock:shadow-root>
          <slot></slot>
        </mock:shadow-root>
        <p>
          IC Loading Indicator Test
        </p>
      </ic-typography>
    </div>
  </mock:shadow-root>
</ic-loading-indicator>
`;

exports[`ic-loading-indicator component should set the typography variant for the label to 'h4' for the 'default' size 1`] = `
<ic-loading-indicator class="label" label="IC Loading Indicator Test" size="default" type="circular">
  <mock:shadow-root>
    <div class="ic-loading-container">
      <div aria-label="Loading" aria-labelledby="ic-loading-label" aria-valuemax="100" aria-valuemin="0" class="ic-loading-circular-outer indeterminate" role="progressbar">
        <div class="ic-loading-circular-inner">
          <svg class="ic-loading-circular-svg" viewBox="0 0 0 0">
            <circle cx="0" cy="0" r="0"></circle>
            <circle cx="0" cy="0" r="0"></circle>
          </svg>
        </div>
      </div>
      <ic-typography class="ic-loading-label ic-typography-h4" id="ic-loading-label" role="status">
        <mock:shadow-root>
          <slot></slot>
        </mock:shadow-root>
        <p>
          IC Loading Indicator Test
        </p>
      </ic-typography>
    </div>
  </mock:shadow-root>
</ic-loading-indicator>
`;

exports[`ic-loading-indicator component should set the typography variant for the label to 'label' for the 'small' size 1`] = `
<ic-loading-indicator class="label" label="IC Loading Indicator Test" size="small" type="circular">
  <mock:shadow-root>
    <div class="ic-loading-container">
      <div aria-label="Loading" aria-labelledby="ic-loading-label" aria-valuemax="100" aria-valuemin="0" class="ic-loading-circular-outer indeterminate" role="progressbar">
        <div class="ic-loading-circular-inner">
          <svg class="ic-loading-circular-svg" viewBox="0 0 0 0">
            <circle cx="0" cy="0" r="0"></circle>
            <circle cx="0" cy="0" r="0"></circle>
          </svg>
        </div>
      </div>
      <ic-typography class="ic-loading-label ic-typography-label" id="ic-loading-label" role="status">
        <mock:shadow-root>
          <slot></slot>
        </mock:shadow-root>
        <p>
          IC Loading Indicator Test
        </p>
      </ic-typography>
    </div>
  </mock:shadow-root>
</ic-loading-indicator>
`;
